packages/core/bootstrap/src/lib/modules/requester.ts

```diff
@@ -219,6 +225,32 @@ export class Requester {
       adapterResponse.providerStatusCode = response.status
     }
 
+    if (parseBool(getEnv('TELEMETRY_DATA_ENABLED'))) {
+      adapterResponse.telemetry = {
+        rateLimitEnabled: parseBool(getEnv('RATE_LIMIT_ENABLED')),
+        wsEnabled: parseBool(getEnv('WS_ENABLED')),
```

Comment:
One big problem with the current approach is that it overlooks the `envDefaultOverrides` for each adapter, since `getEnv` needs the adapter context to get the overrides. An example is `WS_ENABLED`, which is overridden to `true` by default for some adapters.

One option is to pass in the `context` to every `Requester.success` call, although that would get messy. Another option is to set this up as a separate middleware, which I believe should be able to attach the telemetry data to the result object. See [here](https://github.com/smartcontractkit/external-adapters-js/blob/54514ec52436a9a16fb45d05738e76a011a3fd46/packages/core/bootstrap/src/index.ts#L81-L94) for other middleware examples.

Reply:
Hey Austin, thanks for review and comment. Indeed i missed this part. What about another option, to merge envDefaultOverrides with process.env when we are starting the server ?  IDK if it's intentional that we keep them separate. 

```
 const envDefaultOverrides = adapterContext.envDefaultOverrides
 for (const key in envDefaultOverrides) {
   if (!process.env[key]) {
     process.env[key] = envDefaultOverrides[key];
   }
}
```

Reply:
> IDK if it's intentional that we keep them separate.

Not intentional, just a design-by-omission. I am open to the idea of setting the env in a single place when the server is started, but it's unconventional IMO. Let's see what @boxhock thinks.

Reply:
@karen-stepanyan I like the idea, I think that's how it should work instead of checking for base defaults or overrides in each get. Ideally we wouldn't depend on global variables (harder to mock) but it might make sense to make the change easier.

Reply:
Good catch @austinborn. Global variables should generally be avoided IMO, but I do think that this use case warrants this solution. As @alejoberardino says, we should have had something like this for input params as well, but that's way out of scope for this PR (and likely won't be changed until v3 anyways).

TL;DR: I'm good with setting envDefaultOverrides in process.env on startup. Let's make sure that we properly test this though, as this is a change that's relevant regardless of whether the feature flag for telemetry is enabled or not.